pinc/wordcheck_engine.inc

```diff
@@ -785,19 +785,24 @@ function load_word_list($path)
         return [];
     }
 
-    // Note: we're unable to use the file() function's FILE_IGNORE_NEW_LINES
-    // parameter since it wasn't added until PHP 5.0.0 *and* because
-    // it only removes \n, not \r\n, so we'd still need to rtrim later
-    $words = file($path);
-    if ($words === false) {
+    // It's faster to load the entire file into memory, remove any trailing
+    // spaces, and then explode it into an array than use file() to load
+    // it into an array and then rtrim() / rtrim_walk() the entries. Word lists
```

Comment:
A quick check says that we actually have a couple of word lists that are 512K. I need to confirm this is still OK with word lists of that size.

Reply:
I'm comfortable with this being OK and the change worth the memory cost for these larger files.